src/Catalyst.Abstractions/Ledger/IWeb3EthApi.cs

```diff
@@ -21,16 +21,32 @@
 
 #endregion
 
+using Catalyst.Abstractions.Consensus.Deltas;
+using Catalyst.Abstractions.Hashing;
 using Catalyst.Abstractions.Kvm;
+using Catalyst.Abstractions.Ledger.Models;
+using Catalyst.Protocol.Deltas;
+using Catalyst.Protocol.Transaction;
+using Lib.P2P;
+using Nethermind.Core.Crypto;
 using Nethermind.Store;
 
 namespace Catalyst.Abstractions.Ledger
 {
     public interface IWeb3EthApi
     {
-        IDeltaExecutor DeltaExecutor { get; }
         IStateReader StateReader { get; }
         IDeltaResolver DeltaResolver { get; }
-        IStateRootResolver StateRootResolver { get; }
+        IDeltaCache DeltaCache { get; }
+
+        IDeltaExecutor Executor { get; }
+        IStorageProvider StorageProvider { get; }
+        IStateProvider StateProvider { get; }
+        IHashProvider HashProvider { get; }
+
+        Keccak SendTransaction(PublicEntry publicEntry);
+        
+        TransactionReceipt FindReceipt(Keccak transactionHash);
+        bool FindTransactionData(Keccak transactionHash, out Cid deltaHash, out Delta delta, out int index);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider refactoring this method in order to remove the need for this 'out' modifier.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979613)